### ISSUE TITLE
Allow adding manifest entries like Implementation-Title, Implementation-Version

### ISSUE THREAD

**Issue by treimers** [fvarrui/JavaPackager#79]
I'm submitting a feature request on building of Jar file.

- [ ] bug report
- [x] feature request
- [ ] other

**Short description of the issue/suggestion:**
When the jar file is packaged it would be nice to allow adding manifest entries like Implementation-Title, Implementation-Version and others.

Apart from Manifest Version, Classpath or Main-Class a manifest file of a jar file can contain information about application title, version, vendor etc. or other data. These are called package version, a description can be found [here in Oracle docs](https://docs.oracle.com/javase/tutorial/deployment/jar/packageman.html).

The data can be retrieved by application at runtime using methods like

-  getClass().getPackage().getImplementationTitle()
-  getClass().getPackage().getImplementationVersion()
-  getClass().getPackage().getImplementationVendor()

See [Oracles Java doc](https://docs.oracle.com/javase/9/docs/api/java/lang/Package.html#getImplementationTitle--) for these methods.

A manifest file containing the implementation and specification data might look like

```
Manifest-Version: 1.0
Archiver-Version: Plexus Archiver
Created-By: Apache Maven 3.6.3
Built-By: thorsten
Build-Jdk: 15
Specification-Title: JMyApp
Specification-Version: 1.2
Specification-Vendor: Example Tech, Inc.
Implementation-Vendor: Thorsten
Implementation-Title: MyApp
Implementation-Version: 1.2
Main-Class: myApp
```

I think the maven jar plugin is used in this plugin. The maven jar plugin has options to add Manifest entries that can be found in the [official documentation](https://maven.apache.org/shared/maven-archiver/examples/manifestEntries.html).

**Steps to reproduce the issue/enhancement:**

Not applicable

**What is the expected behavior?**

Additional manifest entries can be given somehow like this

```xml
<additionalEntries>
	<entry>
		<name>Name</name>
		<value>java/util/</value>
	</entry>
	<entry>
		<name>Implementation-Title</name>
		<value>App</value>
	</entry>
	<entry>
		<name>Implementation-Version</name>
		<value>1.2</value>
	</entry>
</additionalEntries>
```

**What is the current behavior?**

Not applicable

**What is the motivation / use case for changing the behavior?**

I wolud like to enhance the current software in order to create mainfest with implementation title, version and vendor.

**Please tell us about your environment:**

- JavaPackager version: 1.4.1
- OS version: Mac OS X 10.11.6
- JDK version: OpenJDK Runtime Environment AdoptOpenJDK (build 15+36) 
- Build tool: 
	- [x] Maven 
	- [ ] Gradle 

**Other information** (e.g. related issues, suggestions how to fix, links for us to have context)

Not applicable

**Comment by fvarrui** [comment]
Until we implement this new feature, you can use your own JAR using the `runnableJar` property.
Thanks!

**Comment by fvarrui** [comment]
Hi @treimers!
it's implemented in v1.4.2-SNAPSHOT ... now you can use `additionalManifestEntries` property. Keep in mind that it's a map, so you must use it as follows:
```xml
<additionalManifestEntries>
    <Name>java/util/</Name>
    <Implementation-Title>App</Implementation-Title>
    <Implementation-Version>1.2</Implementation-Version>
</additionalManifestEntries>
```
Thanks for your contribution, as usual!

**Comment by treimers** [comment]
Hi @fvarrui,

you are so fast, this is awesome. Thank you very much!

Sorry to say but I missed something. From what I learned yesterday you can add the specification and implementation data for more than one package. The resulting manifest file look like this

```
Manifest-Version: 1.0
Created-By: peter

Name: foo/
Implementation-Version: foo1

Name: bar/
Implementation-Version: bar1
```

See this discussion at [Stackoverflow](https://stackoverflow.com/questions/32390699/how-to-set-package-version) (scroll to the end of the thread).

This means that the implemented **additionalManifestEntries** is still correct and can be used to add additional properties. But for the story with the package data **manifestSections** must be used. They as well are available in the Maven jar plugin as documented [here](https://maven.apache.org/shared/maven-archiver/examples/manifestSections.html).

Sorry, I did not understand the Manifest options completely when I opened this feature request.

**Comment by fvarrui** [comment]
Ok, now I understand you 😃. You need sections. It will take a bit more time, because I have to implement it for Gradle too, but I think I know how to do it.

**Comment by fvarrui** [comment]
I've just implemented this new feature in v1.4.2-SNAPSHOT.

Config for Maven:
```xml
<manifest>
    <additionalEntries>
        <Created-By>Peter</Created-By>
    </additionalEntries>
    <sections>
        <section>
            <name>foo/</name>
            <entries>
                <Implementation-Version>foo1</Implementation-Version>
            </entries>
        </section>
        <section>
            <name>bar/</name>
            <entries>
                <Implementation-Version>bar1</Implementation-Version>
            </entries>
        </section>
    </sections>
</manifest>
```
and for Gradle:
```groovy
manifest {
    additionalEntries = [
        'Created-By': 'Peter'
    ]
    sections = [
        new io.github.fvarrui.javapackager.model.ManifestSection([
            name: "foo/",
            entries: [
                'Implementation-Version': 'foo1'
            ]
        ])
    ]
}
```
Both produce the same `MANIFEST.MF` file:
```
Manifest-Version: 1.0
Created-By: peter

Name: foo/
Implementation-Version: foo1
```

**Comment by fvarrui** [comment]
v1.4.2 released to Maven Central.
